src/main/kotlin/hwr/oop/core/Deck.kt

```diff
@@ -0,0 +1,45 @@
+package hwr.oop.core
+
+class Deck {
+
+    private val suits = listOf("Hearts", "Diamonds", "Clubs", "Spades")
+    private val ranks = listOf("2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A")
+
+    private val originalOrder: List<Card>
+    private val cards: MutableList<Card>
+
+    init {
+        val tempList = mutableListOf<Card>()
+
+        for (suit in suits) {
+            for (rank in ranks) {
+                val card = Card(suit, rank)
+                tempList.add(card)
+            }
+        }
+
+        originalOrder = tempList.toList()
+        cards = tempList.toMutableList()
+    }
+
+    /*fun peak (): String {
+        for (card in cards) {
+            val cardTitle = (card.rank + " " + card.suit + "\n")
+
+        }
+        return cardTitle
+    }*/
+
+
+    fun peak2 (): List<Card> {
+        return cards
+    }
+
+
+
+    fun deckShuffle(): Boolean {
+        val beforeShuffle = cards.toList()
+        cards.shuffle()
+        return beforeShuffle != cards
+    }

```

Comment:
Comparing specifically by positions changes ensures our wanted behavior...  

```suggestion
    fun shuffle() {
        val beforeShuffle = cards.toList()
        cards.shuffle()

        // Check if shuffle was successful
        val differentPositions = beforeShuffle.zip(cards).count { (before, after) -> before != after }
        if (differentPositions == 0) {
            shuffle()
        }
    }
```